Update doc with use of $ as escape char (#30041)
elastic/beats: libbeat/docs/shared-env-vars.asciidoc

Where `error_text` is custom text that will be prepended to the error
message if the environment variable cannot be expanded.

> If you need to use a special character in your configuration file then you can use `$` to escape the expansion, e.g. `${` or `}` can be escaped with `$${` or `$}`.

Reviewer: ```suggestion
If you need to use a special character in your configuration file, you can use `$` to escape the expansion. For example, you can escape `${` or `}` with `$${` or `$}`.
```

> If you need to use a special character in your configuration file, you can use `$` to escape the expansion. For example, you can escape `${` or `}` with `$${` or `$}`.

Reviewer: ```suggestion
If you need to use a special character in your configuration file, use `$` to escape the expansion. For example, you can escape `${` or `}` with `$${` or `$}`.
```
